src/api/users.rs

```diff
@@ -16,4 +16,71 @@ impl LichessApi<reqwest::Client> {
     ) -> Result<Vec<status::User>> {
         self.get_single_model(request.into()).await
     }
+
+    pub async fn get_rating_history(
+        &self,
+        request: impl Into<rating_history::GetRequest>,
+    ) -> Result<rating_history::RatingHistory> {
+        self.get_single_model(request.into()).await
+    }
+
+    /// Get performance statistics of a user.
+    pub async fn get_user_performance_statistics(
+        &self,
+        request: impl Into<performance::GetRequest>,
+    ) -> Result<performance::Performance> {
+        self.get_single_model(request.into()).await
+    }
+
+    pub async fn get_users_by_id(
+        &self,
+        request: impl Into<by_id::PostRequest>,
+    ) -> Result<Vec<User>> {
+        self.get_single_model(request.into()).await
+    }
+
+    pub async fn get_live_streamers(&self) -> Result<Vec<StreamingUser>> {
+        self.get_single_model(live_streamers::GetRequest::new())
+            .await
+    }
+
+    pub async fn get_crosstable(
+        &self,
+        request: impl Into<crosstable::GetRequest>,
+    ) -> Result<Crosstable> {
+        self.get_single_model(request.into()).await
+    }
+
+    /// Get user autocomplete results.
+    ///
+    /// This differs from [`LichessApi::autocomplete_usernames`] by not returning user information
+    /// and not only the usernames.
+    pub async fn autocomplete_users(
+        &self,
+        request: impl Into<autocomplete::GetRequest>,
+    ) -> Result<Vec<LightUser>> {
+        self.get_single_model(request.into()).await
+    }
+
+    /// Get username autocomplete results.
+    ///
+    /// This differs from [`LichessApi::autocomplete_users`] by not returning any user information
+    /// aside from the username itself.
+    pub async fn autocomplete_usernames(
+        &self,
+        request: impl Into<autocomplete_name::GetRequest>,
+    ) -> Result<Vec<String>> {
+        self.get_single_model(request.into()).await
+    }
```

Comment:
While it is possible to make an untagged enum to resolve this as a single function the type is unfriendly and a bit user hostile, the callers shouldn't have to write code like the below.

```rs
let Autocompletion::Users { users } =
    api.autocomplete(GetRequest::new("foo", None, Some(true)))?
else {
    unreachable!("Should always return Autocompletion::Users");
};
```

If Lichess were to change the query parameter name this may actually reach the other branch, having one method therefore gives less type safety at little benefit. While it may be true that it's easy to pattern match it is very costly from an ergonomics perspective.